polaris-react/src/utilities/event-listener/hooks.ts

```diff
@@ -0,0 +1,21 @@
+import {useEffect, useRef} from 'react';
+
+export type ElementType = Window | Document | HTMLElement;
+
+export function useEventListener(
+  event: string,
+  handler: () => void,
+  element: ElementType = window,
+  options?: AddEventListenerOptions,
+) {
+  const handlerRef = useRef(handler);
+
+  useEffect(() => {
+    const handler = handlerRef.current;
+    element.addEventListener(event, handlerRef.current, options);
+
+    return () => {
+      element.removeEventListener(event, handler, options?.capture);
```

Comment:
```suggestion
      element.removeEventListener(event, handler, options);
```

Looks sufficient to pass the `options` to `removeEventListener`:
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener#syntax

Reply:
I thought I tested with options and it didn't remove. I'll check again